.github/workflows/semgrep-rules-test-develop.yml

```diff
@@ -25,3 +25,11 @@ jobs:
         name: semgrep_runs
         path: semgrep_runs_output.tar.gz
         if-no-files-found: error
+    - uses: zdurham/s3-upload-github-action@master
```

Comment:
Seems a bit risky to rely on a random github action that we're passing our secrets into. Would it be easy enough to just install awscli on a base image that has pip, or maybe we could pin this to a commit hash?

Reply:
Looked into this more, there's also:

https://github.com/eine/tip

where you can use a single pre-release and continuously update it to provide a consistent URL for the artifact 🤔 
but essentially we do run into the problem of this requiring the github token secret being passed in.

Then we wouldn't need to generate our own image with pip & awscli. What do you think?

Reply:
sorry @colleend I wasn't clear about what I meant by pin -- I was thinking pin `zdurham/s3-upload-github-action` to a commit hash, rather than being `@master`, so that we can just audit the code at that commit hash and feel comfortable with it. I think my preferred solution would just be to grab a base image and install AWS CLI, but defer to you.

Reply:
+1 here. We should prefer not to pipe `AWS_SECRET_ACCESS_KEY` to random 3 star Github Action repos.

Reply:
Ah yes, haven't looked at this in a bit -- discussed with Nathan a little, and we were talking about just manually uploading the new semgrep_runs json to r2c.dev until registry 2.0 updates. 

We should probably discuss this a little bit more -- will the registry 2.0 updates get rid of semgrep-runs?